nav2_core/include/nav2_core/behavior.hpp

```diff
@@ -65,6 +73,11 @@ class Behavior
    * @brief Method to deactive Behavior and any threads involved in execution.
    */
   virtual void deactivate() = 0;
+
+  /**
+   * @brief Method to deterimen the required costmap info
+   */
+  CostmapInfoType getResourceInfo() {return CostmapInfoType::LOCAL;}
```

Comment:
Shouldn't this be a virtual function so that it can be overridden for each plugin? I also don't think this is reasonable to have a default, this should be a forced implementation `= 0` for each plugin to setup